src/6model/reprs/VMArray.c

```diff
@@ -49,11 +49,13 @@ static void copy_to(MVMThreadContext *tc, MVMSTable *st, void *src, MVMObject *d
     dest_body->elems = src_body->elems;
     dest_body->ssize = src_body->elems;
     dest_body->start = 0;
+    dest_root->header.flags |= MVM_CF_USES_FSA;
     if (dest_body->elems > 0) {
         size_t  mem_size     = dest_body->ssize * repr_data->elem_size;
         size_t  start_pos    = src_body->start * repr_data->elem_size;
         char   *copy_start   = ((char *)src_body->slots.any) + start_pos;
-        dest_body->slots.any = MVM_malloc(mem_size);
+        dest_root->header.flags |= MVM_CF_USES_FSA;
```

Comment:
This flag is set before the conditional as well as inside of it?

Reply:
Ugh, guess I forgot to remove the one inside the conditional, I'll do that.